java/src/main/java/com/genexus/internet/HttpClientJavaLib.java

```diff
@@ -78,7 +114,13 @@ private static void getPoolInstance() {
 				RegistryBuilder.<ConnectionSocketFactory>create()
 					.register("http", PlainConnectionSocketFactory.INSTANCE).register("https", getSSLSecureInstance())
 					.build();
-			connManager = new PoolingHttpClientConnectionManager(socketFactoryRegistry);
+			String useFirstIpDnsResolver = getGxIpResolverConfig();
+			if (useFirstIpDnsResolver != null) {
+				DnsResolver dnsResolver = new FirstIpDnsResolver();
+				connManager = new PoolingHttpClientConnectionManager(socketFactoryRegistry, dnsResolver);
+			} else {
+				connManager = new PoolingHttpClientConnectionManager(socketFactoryRegistry);
+			}
```

Comment:
Maybe it better to use
boolean useCustomDnsResolver = getGxIpResolverConfig() != null;

PoolingHttpClientConnectionManager connManager = useCustomDnsResolver
    ? new PoolingHttpClientConnectionManager(socketFactoryRegistry, new FirstIpDnsResolver())
    : new PoolingHttpClientConnectionManager(socketFactoryRegistry);